### DISCUSSION TITLE
dynamic viewUrl and pathSegment parameters in the same level as a static set of parameters

### DISCUSSION THREAD

**Discussion by andrew-tud** [SAP/luigi#3023] [Q&A] [answered]
Hi,

I am trying to have a dynamic viewUrl and pathSegment to be on the same navigation level as a set of child navigation parameters as per this config:
```

children: [
                  // this is a child navigation with dynamic parameters :path3
                  {
                    category: {
                      label: 'iFrame 2 c',
                      icon: 'home',
                      collapsible: true
                    }, 
                    pathSegment: ':path3',
                    label: 'dynamic path',
                    viewUrl: 'http://127.0.0.1:8080/:path3',
                    hideFromNav: false,
                    loadingIndicator: {
                      enabled: false
                    },
                  },
                  // this is a child navigation with static parameters. Notice as well that the pathSegment is different from the viewUrl path.
                  {
                    pathSegment: 'iframeD',
                    category: 'iFrame 2 c',
                    label: 'static path',
                    viewUrl: 'http://127.0.0.1:8080/iframe4.html',
                    loadingIndicator: {
                      enabled: false
                    },
                  }
]
}
```
The reason for this is that I need to display some predefined links in the MFE Shell side navigation in the same level as the dynamic navigation. With the above config, the requirement can work if the static pathSegment is the same as the static viewUrl, but when I try to have a different user friendly name for the static pathSegment, the viewUrl cannot become different from the static pathSegment as specified in the above config. Is there a way to do this?

It is a minor functionality that might come up being asked in our POC experiment but I am just wondering if I can do extra steps to achieve this.

**Comment by ndricimrr** [comment, marked as answer]
Hi @andrew-tud ,

The current behaviour you are asking for :
```
{
        // parentNode
        pathSegment: "parent",
        children: [
             { // dynamic child },
             { // static child }
        ]
}
```
to have a dynamic and a static child on the same level is something that we do not allow on purpose. The situation that is prohibited here is the following:

<img width="280" alt="Screenshot 2022-12-05 at 16 56 19" src="https://user-images.githubusercontent.com/18335489/205682455-523bfdd1-6947-4785-b493-8328a89778b8.png">
As one of the children will be static, when opening the parent you will be displaying a left navigation item which will have an empty value in the beginning. For that reason we do not allow that behaviour. 

Could you give us some more information on your specific use case scenario and what exactly you are trying to achieve and why. It could maybe help us provide/suggest with a better approach to this ?
